### PR TITLE
Fix 1124: provide clear naming for cache directories

### LINKED ISSUE #1124: Provide clear naming for the inconsistent ways to get the cache directory
```python
import openml
openml.config.get_cache_directory()   # is:   '~\\.openml\\org\\openml\\www'
openml.config.cache_directory         # is:   '~\\.openml'
``` 

`get_cache_directory`  gets the cache for the currently configured server, while `cache_directory` gets the root of the cache directory.
`set_cache_directory(str)` configures the root.

If we plan to maintain an external interface for both types, I propose we:
 - we make them both functions so they are accessed in the same way (and using a function will make a get\set pattern easier where a set can be optionally for the session or saved to the configuration file)
 - make it clear in the name what (part of) the cache directory is returned

### PR DESCRIPTION
**Pull request by mfeurer**
#### Reference Issue
Fixes #1124

#### What does this PR implement/fix? Explain your changes.
* Introduce the concept of a `root_cache_directory`, i.e. the cache directory in which the concrete cache directories for the different servers reside
* Rename the function `set_root_directory` to `set_root_cache_directory`
* Improve the docstring of both functions

#### How should this PR be tested?
Regular unit tests.

#### Any other comments?
No.

### CODE REVIEW

File: openml/config.py [hunk 1]
```diff
@@ -351,27 +362,32 @@ def get_cache_directory():
...
+def set_root_cache_directory(root_cache_directory):
+    """Set module-wide base cache directory.
 
-    Sets the cache directory into which to download datasets, tasks etc.
+    Sets the base cache directory that defines how the actual cache
+    directory for the server being used is derived. This is
+    ``root_cache_directory / top-level domain / second-level domain /
+    hostname``, and by default is set to
+    ``root_cache_directory / org / openml / www`` for the standard
+    OpenML.org server.
```
Comment (PGijsbers) [inline openml/config.py:377]:
```suggestion
    Sets the root cache directory wherein the cache directories are created to store content
    from different openml servers. For example, by default, cached data for the standard openml.org
    server is stored in ``root_cache_directory / org / openml / www``.
```
I found the description a bit vague, due to inconsistent use of base vs. root and that it does not really define how the cache directory is derived, rather which directory the caches are placed in. This is just a suggestion, but any update the removes this ambiguity is fine with me.

File: openml/testing.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def setUp(self, n_levels: int = 1):
         self.production_server = "https://openml.org/api/v1/xml"
         openml.config.server = TestBase.test_server
         openml.config.avoid_duplicate_runs = False
-        openml.config.cache_directory = self.workdir
+        openml.config._cache_directory = self.workdir
```
Comment (PGijsbers) [inline openml/testing.py:96]: We should probably use `set_root_cache_directory()` here.

**Review by PGijsbers — CHANGES_REQUESTED** [review]
See comments.

File: openml/config.py (hunk 1 above)
Comment (mfeurer) [reply openml/config.py]: Changed this, thanks for spotting that I forgot to replace `base` here with `root`.

**Review by PGijsbers — APPROVED** [review]
LGTM

**Pull request merged**